service/history/historyEngine.go

```diff
@@ -2246,8 +2242,21 @@ func (e *historyEngineImpl) ResetWorkflowExecution(
 	}
 	if request.GetWorkflowTaskFinishEventId() <= common.FirstEventID ||
 		request.GetWorkflowTaskFinishEventId() >= baseMutableState.GetNextEventID() {
-		return nil, serviceerror.NewInvalidArgument("Workflow task finish ID must be > 1 && <= workflow next event ID.")
+		return nil, serviceerror.NewInvalidArgument("Workflow task finish ID must be > 1 && <= workflow last event ID.")
```

Comment:
probably the message should state:
"< workflow last event ID."

WorkflowTaskID being the same as NextEventID is an error.

Reply:
the message says last event ID (not NextEventID), so using <= is correct.